File: 01-09/Problem_9/Problem9.java

public class Problem9 {

    public static void main(String[] args) {
        int c = 1;
        
        while (true) {
            for (int b=1; b < c; b++) {
                for (int a=1; a < b; a++) {
                    if (a+b+c==1000 && a*a+b*b == c*c) {
                        System.out.println(a*b*c);
                        return;
                    }  
                }
            }
            c++;
        }
    }
}

File: 01-09/Problem_9/problem9.pl
problem9(Answer) :-
	between(1,1000,A),
	Bmin is A+1, Bmax is (1000-A),
	between(Bmin,Bmax,B),
	C is 1000-A-B,
	A*A + B*B =:= C*C,
	Answer is A*B*C, !.
